core/kernel.rbs

```diff
@@ -488,7 +488,7 @@ module Kernel : BasicObject
   #     Hash(nil)              # => {}
   #     Hash([])               # => {}
   #
-  def self?.Hash: [K, V] (Object x) -> ::Hash[K, V]
+  def self?.Hash: [K, V] (nil | [] | _ToHash[K, V])  -> Hash[K, V]
```

Comment:
> I was debating on whether I should make a separate `(nil | []) -> {}` case, but I decided against it.

No need to, at all, since `nil` and empty arrays already have `#to_h`, let their own RBSs do the `[K, V]` magic. (I’ll check RBS for `NilClass#to_h`)

```suggestion
  def self?.Hash: [K, V] (_ToHash[K, V])  -> Hash[K, V]
```

Reply:
`NilClass#to_h` gives `Hash[untyped, untyped]` :+1:

Reply:
It's not the exact same @ParadoxV5—`_ToHash` requires `to_hash`, not `to_h`. So the `nil | []` case is distinct.